src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp

```diff
@@ -722,6 +724,18 @@ QueryPlanPtr MergeTreeDataSelectExecutor::readFromParts(
                 max_partitions_to_read);
     }
 
+    String query_id;
+    if (data_settings->max_concurrent_queries > 0)
+    {
+        if (data_settings->min_marks_to_honor_max_concurrent_queries > 0
```

Comment:
Check `data_settings->min_marks_to_honor_max_concurrent_queries > 0` is not needed because `sum_marks >= 0` always as unsigned.